Gems/Atom/Feature/Common/Code/Source/Mesh/MeshFeatureProcessor.cpp

```diff
@@ -1026,6 +1022,118 @@ namespace AZ
+                {
+                    // Currently GetAverageColor() is only implemented for a StreamingImage
+                    auto baseColorStreamingImg = azdynamic_cast<RPI::StreamingImage*>(baseColorImage.get());
+                    if (baseColorStreamingImg)
+                    {
+                        // Note: there are quite a few hidden assumptions in using the average
+                        // texture color. For instance, (1) it assumes that every texel in the
+                        // texture actually gets mapped to the surface (or non-mapped regions are
+                        // colored with a meaningful 'average' color, or have zero opacity); (2) it
+                        // assumes that the mapping from uv space to the mesh surface is
+                        // (approximately) area-preserving to get a properly weighted average; and
+                        // mostly, (3) it assumes that a single 'average color' is a meaningful
+                        // characterisation of the full material.
+                        Color avgColor = baseColorStreamingImg->GetAverageColor();
+
+                        // We do a simple 'multiply' blend with the base color for now. Warn
+                        // the user if something else was intended.
+                        propertyIndex = material->FindPropertyIndex(AZ::Name("baseColor.textureBlendMode"));
+                        if (propertyIndex.IsValid())
+                        {
+                            AZ::Name textureBlendMode = material->GetMaterialPropertiesLayout()
+                                     ->GetPropertyDescriptor(propertyIndex)
+                                     ->GetEnumName(material->GetPropertyValue<uint32_t>(propertyIndex));
+                            if (textureBlendMode != AZ::Name("Multiply"))
+                            {
+                                AZ_Warning("MeshFeatureProcessor", false, "textureBlendMode '%s' is not "
+                                        "yet supported when requesting BaseColor irradiance source, "
+                                        "defaulting to 'Multiply'.", textureBlendMode.GetCStr());
+                            }
+                        }
+                        // 'Multiply' blend mode:
+                        subMesh.m_irradianceColor = avgColor * subMesh.m_baseColor;
+                    }
+                    else
+                    {
+                        AZ_Warning("MeshFeatureProcessor", false, "Using BaseColor as irradianceColorSource "
+                                "is currently only supported for textures of type StreamingImage");
+                        // Default to the flat base color
+                        subMesh.m_irradianceColor = subMesh.m_baseColor;
+                    }
+                }
+                else
+                {
+                    // No texture, simply copy the baseColor
+                    subMesh.m_irradianceColor = subMesh.m_baseColor;
```

Comment:
I wonder if we should have a third color mode "BaseColorTint" that ignores the texture and just uses baseColor.color.